Improve burtin.py appearance

bryevdv opened the pull request:
- [x] issues: fixes #13116 

Fixes the text format issue by hard-coding the labels. Also adds `match_aspect=True` and moves the "circular" axes underneath the radial bars for for improved legibility. Now looks like:

<img width="766" alt="Screenshot 2023-05-10 at 21 03 16" 
src="https://github.com/bokeh/bokeh/assets/1078448/d2bf6e39-701f-40f6-b460-a5579f81adbb">

@mattpap if you know why these text labels with middle baseline are not lining up with the circles of the same radius I would love to fix that too:
<img width="536" alt="Screenshot 2023-05-10 at 21 04 02" src="https://github.com/bokeh/bokeh/assets/1078448/6ab8a527-090d-4626-8b9a-e6fc6d0e71c8">

bryevdv committed:
Simplify and improve burtin.py

ianthomas23 commented on examples/topics/pie/burtin.py:
```diff
@@ -64,6 +64,13 @@ def rad(mic):
     0, 0, inner_radius, outer_radius, -big_angle+angles, angles, color=colors,
 )
 
+# circular axes and lables
```
```suggestion
# circular axes and labels
```

mattpap commented:
> if you know why these text labels with middle baseline are not lining up with the circles of the same radius I would love to fix that too:

Removing the toolbar fixes this. Given the plot has fixed ranges, then `match_aspect` does nothing. I would either remove the toolbar or use `inner_width` and `inner_height`. I would also replace `text_align` and `text_baseline` with the new `anchor=value("center")`.

bryevdv commented:
Ah, I completely skipped over the fixed range. There's actually quite a few places this example can be modernized e.g. color mappers and transforms. 

Auto-ranging doesn't *quite* work for this example. Angled text labels at the top get cut off. Could add an invisible placeholder glyph.

bryevdv commented:
re:

> new `anchor=value("center")`

It's probably worth going through all the examples to update to best latest practices. I can do that in this PR. @mattpap can you provide a quick summary or table for conversions.

bryevdv commented:
Made some changes to simplify and improve the appearance and legibility at the same time, and maybe I don't completely hate this example any more
<img width="797" alt="Screenshot 2023-05-13 at 20 11 52" src="https://github.com/bokeh/bokeh/assets/1078448/24259d0c-87a0-4f53-923a-fe8704d7021c">

Edit: to be explicit, at this point I'd rather stay true to the general spirit of "reproduction of Burtin" but make room for improvements, or else just delete the example altogether.

bryevdv commented:
Needs new thumbnails, but I don't intend to make more code changes

bryevdv committed:
add new numbnails

bryevdv committed:
fix sampledata ref

bryevdv committed:
fix plausible config

bryevdv committed:
fix factors/cmap mismatch

bryevdv commented:
I've fixed a few more small docs issues here for the sake of expediency. I believe I've also fixed the issue w/ Plausible not loading. Per the [theme docs](https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/analytics.html#plausible-analytics) the analytics configs go in their own separate sub-dict in the config. cc @tcmetzger @pavithraes

bryevdv commented:
Just noting some other issues (not fixed):

@mattpap when I build the docs I see
```
Errors encountered during validation: ["E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text_font='helvetica' [no close matches] {renderer: GlyphRenderer(id='p9243', ...)}"]
``` 
for `theme_glyphs.py` but I don't see it when I run manually. 🤷 Changing to `{"value": "helvetica"}` can  "fix" the issue but that's not a real solution. 

In the near future we should add `BOKEH_VALIDATION_LEVEL=all` to the docs build to make catching validation problems easier.

bryevdv commented:
@bokeh/core Numpy 1.25 was released yesterday and new deprecations are causing min-deps tests to fail:
```
FAILED tests/unit/bokeh/models/test_sources.py::TestColumnDataSource::test_init_groupby_arg - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
FAILED tests/unit/bokeh/models/test_sources.py::TestColumnDataSource::test_data_accepts_groupby_arg - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
FAILED tests/unit/bokeh/models/test_sources.py::TestColumnDataSource::test_init_groupby_data_kwarg - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
FAILED tests/unit/bokeh/models/test_sources.py::TestColumnDataSource::test_init_groupby_with_None_subindex_name - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
FAILED tests/unit/bokeh/models/test_sources.py::TestColumnDataSource::test_data_accepts_groupby_with_None_subindex_name - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
```
I guess we were not actually pinning numpy to the min version 1.16 so I will attempt to do that here to get this passing and merged for a last rc

I couldn't get a conda env working with specific older np versions lie 1.16 so I just used `<1.25` for now. We'll want to examine this on 3.3 branch (and maybe time to raise np min version anyway)

bryevdv committed:
update switcher.json

bryevdv committed:
only MM YY for release date

bryevdv committed:
numpy < 1.25
